### PR TITLE
Include web in built package.

### LINKED ISSUE #217: When running the deploy CI flow, no artifact for the web package is produced.
As stated in the title, the `rust-bindings`, `common` and `nodejs` packages are included in the artifact produced when running the deploy CI flow in github actions. The `web` package is missing however.

Furthermore, it would make sense to include all files needed to publish, to be able to publish from the artifact rather than having to move the build to the local project to release from there.

### PR DESCRIPTION
**Pull request by soerenbf**
## Purpose

Fixes #217 

## Changes

- Caches yarn dependencies
- Include all packages in `build-release`, slim down `rust-bindings` to only include needed files due to difficulties properly publishing with `pkg/*/.gitignore` present.

### CODE REVIEW

**Review by orhoj — DISMISSED** [review]

File: .github/workflows/deployment.yml [hunk 1]
```diff
@@ -56,12 +66,11 @@ jobs:
         with:
           name: build-release
           path: |
-            ./node_modules/@concordium
```
[the remark this answers is not in the document]
Comment (soerenbf) [reply .github/workflows/deployment.yml:59]: It seems like it was just copied from the `build-debug` from `pipeline.yml`, which also did not have the web package included as it's just used for testing.

It's not needed as it is included in the `cache-yarn` cache.

File: .github/workflows/deployment.yml [hunk 2]
```diff
@@ -45,7 +45,17 @@ jobs:
         with:
           version: 'latest'
 
+      - name: Cache dependencies
+        id: yarn-cache
+        uses: actions/cache@v3
+        with:
+          path: node_modules
```
[the remark this answers is not in the document]
Comment (soerenbf) [reply .github/workflows/deployment.yml:52]: yarn install is also called if cache reports a hit (see line 57)

[the remark this answers is not in the document]
Comment (soerenbf) [reply .github/workflows/deployment.yml:52]: The cache action includes a postcache action pr default that handles caching the files included in the cache at the end of the job

File: .github/workflows/deployment.yml [hunk 3]
```diff
@@ -45,7 +45,17 @@ jobs:
...
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn
```
[the remark this answers is not in the document]
Comment (soerenbf) [reply .github/workflows/deployment.yml:55]: Changed it to be the same as the `key` property. It's only used to share resources between `build` and `build-typedoc` jobs.

File: .github/workflows/deployment.yml [hunk 4]
```diff
@@ -45,7 +45,17 @@ jobs:
         with:
           version: 'latest'
 
+      - name: Cache dependencies
```
Comment (shjortConcordium) [inline .github/workflows/deployment.yml:48]: `actions/setup-node@v3` already caches the dependencies, so why do we need to do this? :thinking:

Comment (soerenbf) [reply .github/workflows/deployment.yml:48]: Symlinks to the packages built are not included in the node cache, and these are needed for the typedoc build. The way this worked prior to this change, was to include the symlinks in the artifact meant for publishing, which poluted the `build-release` artifact.

**Review by rasmus-kirk — APPROVED** [review]
LGTM

**Pull request merged**